random-distribution-helpers.go

```diff
@@ -14,60 +14,66 @@ type RandomDistribution interface {
 	String() string
 }
 
-var _ RandomDistribution = RandomConstDistribution{}
+var _ RandomDistribution = &RandomConstDistribution{}
 
 type RandomConstDistribution struct {
 	value time.Duration
 }
 
-func (constDist RandomConstDistribution) Sample() time.Duration {
+func (constDist *RandomConstDistribution) Sample() time.Duration {
 	return constDist.value
 }
 
-func (constDist RandomConstDistribution) String() string {
-	return fmt.Sprintf("Constant value: %d.", constDist.value)
+func (constDist *RandomConstDistribution) String() string {
+	return fmt.Sprintf("Constant value: %v ms.", int64(constDist.value/time.Millisecond))
```

Comment:
For printing time.Duration values, just use %v (without dividing the value, or adding any "ms" suffix yourself). In general: Almost always just use %v when printing anything.

Reply:
It prints the value as nanoseconds which looks ugly. I'd like to have it in milliseconds

Reply:
fmt.Println(3 * time.Minute + 3 * time.Second + 3 * time.Millisecond)
prints:
"3m3.003s"

You probably used "%d" as format character, which prints it as a number, without using the String() method defined on the time.Duration type.

Reply:
Yes, done